lib/mix/lib/mix/scm/git.ex

```diff
@@ -229,4 +236,13 @@ defmodule Mix.SCM.Git do
     {int, _} = Integer.parse(string)
     int
   end
+
+  defp lock_outdated?(opts) do
+    case opts[:lock] do
+      {:git, repo, _rev, lock_opts} ->
+        not git_repos_match?(repo, opts[:git]) or lock_opts != get_lock_opts(opts)
```

Comment:
what if we call it `lock_up_to_date?` so we can write it as:

    git_repos_match?(repo, opts[:git]) and lock_opts == get_lock_opts(opts)

I think it will be clearer than the code today? Thoughts?